src/api/app/models/staging_workflow.rb

```diff
@@ -32,6 +32,33 @@ def ignored_requests
     BsRequest.none # TODO: define this method
   end
 
+  def to_xml
+    builder = Nokogiri::XML::Builder.new
+    staging_workflow_attributes = { project: project.name,
+                                    who: User.current_login,
+                                    when: updated_at }
+
+    builder.staging_workflow(staging_workflow_attributes) do |builder_staging_workflow|
+      staging_projects.each do |staging_project|
+        builder_staging_workflow.staging_project(name: staging_project.name) do |builder_staging_project|
+          staging_project.staged_requests.each do |staged_request|
+            request_attributes = { number: staged_request.number,
+                                   state: staged_request.state,
+                                   creator: staged_request.creator }
+
+            builder_staging_project.request(request_attributes)
+          end
+        end
+      end
+    end
+
+    builder.to_xml(save_with: Nokogiri::XML::Node::SaveOptions::NO_DECLARATION | Nokogiri::XML::Node::SaveOptions::FORMAT)
```

Comment:
Why don't you do all of this in a view?

Reply:
I mean doing the layout of the xml in a `app/views/staging_workflow.xml.builder` instead of using builder inside the model. then we can also make use of this for showing etc.

Reply:
uhhmmm, I will try it :smile_cat:. Thanks

Reply:
It the better pattern for the xml rendering IMHO. see for instance the status/required checks...

Reply:
> but rendering a view in a model 

Hey would you so that? You would also not build HTML inside a model, would you?

Reply:
> but rendering a view in a model 

Hey would you so that? You would also not build HTML inside a model, would you?

Reply:
Okey, I changed the code to use `CanRenderModel` mixing instead of `Nokogiri`